mathics/core/symbols.py

```diff
@@ -435,16 +440,24 @@ def get_head(self) -> "Symbol":
     def get_head_name(self):
         return "System`Symbol"
 
-    def has_changed(self, definitions) -> bool:
+    def is_uncertain_final_definitions(self, definitions) -> bool:
         """
-        Used in Expression.evaluate() to determine if we need to reevaluate
-        an expression.
+        Used in Expression.do_format() to determine if we need to
+        (re)evaluate an expression.
+
+        Here, we have to be pessimistic and return True.
+
+           InputForm[Context[]] == "Global`"
+
+        is an example where we need to evaluate "Global`" and it doesn't start out as a
```

Comment:
@rocky, do you mean the lhs or the rhs?

Reply:
I just checked again, and I see what I have is wrong. The problem is that in evaluating ``$System`Context`` we need another evaluation. 

I will revise the comment in the next iteration.

Reply:
@mmatera Ok - revised. See if this is correct now.